docs/daemon/configuration/genesis.mdx

```diff
@@ -0,0 +1,158 @@
+
+```json title="sample genesis.json file"
+{
+  "genesis_time": "2023-08-28T22:24:04.513472Z",
+  "chain_id": "kwil-chain-RQMl2F",
+  "initial_height": "0",
+  "validators": [
+    {
+      "address": "6F129A1818CB42DD80FFE51943EAC0BC9FF9E4EE",
+      "pub_key": {
+        "type": "tendermint/PubKeyEd25519",
+        "value": "L737LDRTYv7P5I2apSxOBwJD3U0CS7b2eo9K37s2lPM="
+      },
+      "power": "1",
+      "name": "validator-0"
+    }
+  ],
+  "app_hash": "4F8735846B8CFB01A277430A392DD3DB7B66510C358C5F4A2AF88A348CB20A62"
+}
+```
+
+## Genesis file contents
+
+A Genesis file essentially consists of:
+
+### Genesis time
+
+ `genesis_time`: This field specifies the official time at which the blockchain is initialized.
+
+### Chain ID
+
+ `chain_id`: This field specifies the ID of the blockchain network. This must be unique for every blockchain. The ChainID must be less than 50 symbols.
+
+### Initial height
+
+ `initial_height`: This field sets the initial height the blockchain should begin.  `Default value is 0`, but in the scenario of network upgrade or restart, starting from the stopped height ensures uniqueness to previous heights.
+
+### Genesis Validators
+
+ `validators`: This field sets the list of the initial set of validators. Each validator should provide the below information:
+  - `pub_key`: consisting of key types and public key bytes.
+  - `address`: address of the pub_key
+  - `power`: The validator’s voting power, `default: 1`. Power 0 indicates that the node is a non-validator.
+
+Cryptographic keys for a kwild node can be generated and viewed using the `kwil-admin key` command line utility as shown below. 
+
+```bash
+# Generate private key
+kwil-admin key gen -o "path/to/private/key/file"
+
+# View Generated key info
+kwil-admin key info -k "path/to/private/key/file"
+```
+
+### Initial app_hash
+
+ `app_hash`: The app_hash represents the initial application state. If the node is booting up with initial data, the app_hash of the genesis file can be initialized by running the below command. 
+
+  ```bash
+  kwil-admin setup genesis-hash "path/to/initial/db/dir" -g "path/to/my/genesis.json"
+  ```
+
+  If `--genesis or -g` is not provided, the command would stdout the app_hash. 
+
+## Advanced Configurations
+
+### Consensus Params
+
+Genesis file can also be used to establish a predefined set of consensus rules such as:
+#### Block rules:
+
+  - `block`: 
+      - `max_bytes`: This field specifies the maximum block size, in bytes.
+      - `max_height`: This field specifies the maximum gas per block.
+
+#### Validator type:
+
+  - `validator`:
+    -  `pub_key_types`: This field sets the public key types validators can use.
+    -  `join_expiry`: This field sets the number of blocks after which the validator join request expires if not approved. The default expiry is 86400 blocks which is equivalent to 1 day if blocks are produced at the rate of 1 block/sec
+
+#### ABCI app Version:
```

Comment:
Please revisit this entire section on grammar, consistent punctuation, and consistent capitalization of things like title names.